app/workers/mails.php

```diff
@@ -69,6 +65,36 @@ public function run(): void
         }
     }
 
+    protected function getMailer(array $smtp): PHPMailer
+    {
+        $mail = new PHPMailer(true);
+
+        $mail->isSMTP();
+
+        $username = $smtp['username'];
+        $password = $smtp['password'];
+
+        $mail->XMailer = 'Appwrite Mailer';
```

Comment:
Can we overwrite the mailer object from our DI by injecting the current project object when creating it? If project has SMTP setting we use them, if not we use the default Appwrite ones.

Reply:
Didn't find a way to do it, and didn't find any similar pattern as well.

Reply:
You can see an example in the user project. See how we set `project` as a dependency before creating the `user` dependency. 

https://github.com/appwrite/appwrite/blob/25d94f0995bdde80653d4dfcddf4597c11cfedce/app/init.php#L798

Reply:
Hey @eldadfux I'm still confused, this is Worker and we don't yet have DI in Worker?

Reply:
Keeping it in backlog